pdns/arguments.cc

```diff
@@ -365,23 +387,39 @@ void ArgvMap::parseOne(const string &arg, const string &parseOnly, bool lax)
     {
       if(incremental)
       {
-        if(params[var].empty())
+        if(d_params[var].empty())
         {
           if(!d_cleared.count(var))
-            throw ArgException("Incremental parameter '"+var+"' without a parent");
-          params[var]=val;
+            throw ArgException("Incremental setting '"+var+"' without a parent");
+          d_params[var] = val;
         }
         else
-          params[var]+=", " + val;
+          d_params[var] += ", " + val;
       }
       else
       {
-        params[var]=val;
+        d_params[var] = val;
         d_cleared.insert(var);
       }
     }
-    else if(!lax)
-      throw ArgException("Trying to set unknown parameter '"+var+"'");
+    else
+    {
+      // unknown setting encountered. see if its on the ignore list before throwing.
+      vector<string> parts;
```

Comment:
Why not a set? saves you a loop below

Reply:
set does not actually work with stringtok:

```
./misc.hh:102:17: error: no member named 'push_back' in 'std::__1::set<std::__1::basic_string<char>,
      std::__1::less<std::__1::basic_string<char> >, std::__1::allocator<std::__1::basic_string<char> > >'
      container.push_back (in.substr(i));
      ~~~~~~~~~ ^
arguments.cc:409:7: note: in instantiation of function template specialization 'stringtok<std::__1::set<std::__1::basic_string<char>,
      std::__1::less<std::__1::basic_string<char> >, std::__1::allocator<std::__1::basic_string<char> > > >' requested here
      stringtok(parts, d_params["ignore-unknown-settings"], " ,\t\n\r");
      ^
```

Reply:
Bah, I don't see an immediate solution to this, `vector` and `set` are a bit too different...
You still might want to use `std::find` for the searching though

Reply:
I've added a stringtok template variant for set.

Reply:
Since `vector` actually both has `push_back` and `insert`, will the first template be selected in that case?

Reply:
I don't think the `insert` variant would compile if it picked `vector`?

Reply:
`vector`s `insert` shouldn't compile, and IIRC non-compiling templates are not picked if there are other variants?

I haven't seen any trait types that would be useful here.

Reply:
I was reading the template chapters of the C++ reference, but I couldn't find a clear description of the case at hand between all the thousands of words. The build does not seem to agree now though, but I did not look in detail.

Reply:
Yeah, for some reason now `ld` also hates it locally. Which is weird, because that should never be a symbol problem!?
I'll remove it.